### ISSUE TITLE
Market Store

### ISSUE THREAD

**Issue by andreareginato** [lelylan/lelylan#33]
Lelylan Market store must be able to collect an high number of services coming from outside. We shouldn't limit the capability of third party to create new solutions.  This means that somebody add an application to the market store by its own and we will simple validate the acceptance of the application. This should be pretty flexible and let us give a minimal control. A small inspiration can be given by the add on console of heroic. It will be quite important to have different type of organizations (stars, new, categories). Learn more from existing stores.

Actually an app on the market store is a registered app with few more logics. When clicking on the application icon you go to the OAuth page where you grant the access for the specific application. Together with this, the only difference lives in the capability of the market store. You do not change anything. The app automatically set a set of webhooks for the application through the APIs. This means that the app store is just an interface to the accepted applications. It will be the new authorized applications page, with some more features.

When the user accepts the application you get a confirmation that the process has been successfully added. At this point you have a link to the dashboard and you can highlight the new link to the visual part of the application if existing. 

And here comes the visual part. An application can be made in two ways. The first one logical and the second one is visual. The logical one will only use the webhooks. The visual one will give you a new User Experience. This user experience lives in another website and we should not care about this. But here raises a linking problem. We need to uniform the User interface, as little as we can, but giving a fast way to move between all existing APPS.

Actually we can give an header. This header contains all common links like

* Home
* Admin
* Logout
* Applications

Between those you see applications. This page will redirect the user to the authorized apps (filtered through visualization). Through this page I'll be able then to move to other apps. This is a 2 click system and it would be nice to have a list of all accessible UI on the top menu. For the first step keep using common parts.

This system should be also available to build a single (or defined set) of products market. Here the thing starts becoming complicated as we set a number of applications only for a set of devices (each app can have all, some devices, and it should be listed on the app page). 

This means that when I'm in the dashboard of Lelylan I see all the apps and for each app all the devices associated. But when I access the market store defined for another product, I should filter only the apps for that market store and for the specific device(s).

Think at this scenario. I've Lelylan where I've plenty of devices and plenty of applications added. For each application I've a list of devices (probably this list is given from a set of tokens, also for the same app? Do not overcomplicate!!) Then I go to light.me. In here I register to Lelylan and add my light. Then I go to the market place and I see my light. In here I see also the possible applications I can use for this light and I decide which ones to add. This means that I do not see any of the other applications in here, but when I go back to Lelylan I see also this device with the added applications.

The market store definition is quite critical and complex. Before starting making it take at least a couple of days and split its definition in several mailstones. We need to be able to provide a basic service, and then we can add more and more functionalities. This implementation requires at least a couple of monist.
